packages/block-editor/src/hooks/color-panel.js

```diff
@@ -70,7 +61,36 @@ export default function ColorPanel( {
 		}
 
 		setDetectedBackgroundColor( backgroundColor );
-	} );
+
+		// If colors are defined, overwrite detected colors with the defined colors.
+		if ( definedColors.length ) {
+			definedColors.forEach( ( element ) => {
+				if ( 'Background' === element.label ) {
+					if ( element.colorValue ) {
+						setDetectedBackgroundColor( element.colorValue );
+					}
+				} else if ( 'Text' === element.label ) {
+					if ( element.colorValue ) {
+						setDetectedColor( element.colorValue );
+					}
+				} else if ( 'Link' === element.label ) {
+					if ( element.colorValue ) {
+						setDetectedLinkColor( element.colorValue );
+					}
+				}
```

Comment:
```suggestion
				if ( ! color.colorValue ) {
					return;
				}

				if ( 'Background' === color.label ) {
					setDetectedBackgroundColor( color.colorValue );
				}

				if ( 'Text' === color.label ) {
					setDetectedColor( color.colorValue );
				}

				if ( 'Link' === color.label ) {
					setDetectedLinkColor( color.colorValue );
				}
```
There's quite a bit of unnecessary nesting here and the `element` variable name doesn't seem to suit well. Perhaps we could clean that up?